azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -418,6 +418,37 @@ func resourceArmContainerGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"dns_config": {
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nameservers": {
+							Type:     schema.TypeList,
+							Required: true,
+							ForceNew: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"search_domains": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"options": {
+							Type:         schema.TypeString,
```

Comment:
Does `options` allow multiple values (e.g. separated by space)? If so, can we also change it into `schema.TypeSet` here?

Besides, shall we mark it as `Optional` as it is an option itself?

Reply:
I think it does I was just trying to stay as close as possible to the AzureRM go client impl, will change :)
```go
// DNSConfiguration DNS configuration for the container group.
type DNSConfiguration struct {
	// NameServers - The DNS servers for the container group.
	NameServers *[]string `json:"nameServers,omitempty"`
	// SearchDomains - The DNS search domains for hostname lookup in the container group.
	SearchDomains *string `json:"searchDomains,omitempty"`
	// Options - The DNS options for the container group.
	Options *string `json:"options,omitempty"`
}

```

Reply:
Yeah, it is weired why Swagger defines that field as a string as it is actually a space delimited list. I prefer using a Set here to serve better user experience.

Reply:
I changed them both to set types